libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/adapter/InputAdapter.kt

```diff
@@ -30,6 +31,11 @@ internal fun List<NamedType>.inputAdapterTypeSpec(
       .addSuperinterface(KotlinSymbols.Adapter.parameterizedBy(adaptedTypeName))
       .addFunction(notImplementedFromResponseFunSpec(adaptedTypeName))
       .addFunction(writeToResponseFunSpec(context, adaptedTypeName))
+      .apply {
+        if (this@inputAdapterTypeSpec.any { it.deprecationReason != null }) {
+          addAnnotation(suppressDeprecationAnnotationSpec)
+        }
```

Comment:
Would it make sense to also add OptIn if any field is marked as such, like in #4470?

Reply:
Yep, good catch 👍

Reply:
done